src/agent/src/uevent.rs

```diff
@@ -230,6 +230,7 @@ pub async fn watch_uevents(
 
 // Used in the device module unit tests
 #[cfg(test)]
+#[allow(clippy::needless_return)]
```

Comment:
Why not actually fix the needless return instead?

Reply:
> 

I tried to remove the return of this test function. But could not keep the original code logic.

Reply:
How about replacing `return` with `break`?

Reply:
Thanks, but for_each block couldn't use break. I have to add `#[allow(clippy::needless_return)]`

Reply:
@YchauWang What's your rust version? I can't get the same error as you.
And the usage of `for_each` is wrong here, we can use `filter` function to get the matched watcher to avoid using `for_each`.

Reply:
```
root@david-vm:~/devspaces/k3s-build-images# rustc -V
rustc 1.54.0 (a178d0322 2021-07-26)
```